javascript/src/components/streams/StreamControls.jsx

```diff
@@ -0,0 +1,86 @@
+/* global jsRoutes */
+
+'use strict';
+
+var React = require('react/addons');
+var DropdownButton = require('react-bootstrap').DropdownButton;
+var MenuItem = require('react-bootstrap').MenuItem;
+var ButtonGroup = require('react-bootstrap').ButtonGroup;
+var StreamForm = require('./StreamForm');
+var PermissionsMixin = require('../../util/PermissionsMixin');
+var UsersStore = require('../../stores/users/UsersStore');
+
+var StreamControls = React.createClass({
+    mixins: [PermissionsMixin],
+    getInitialState() {
+        return {};
+    },
+    _onResume(evt) {
+        this.props.onResume(this.props.stream);
+        this.refs.dropdownButton.setDropdownState(false);
+    },
+    _onPause(evt) {
+        this.props.onPause(this.props.stream);
+        this.refs.dropdownButton.setDropdownState(false);
+    },
+    _onEdit(evt) {
+        this.refs.streamForm.open();
+    },
+    _onClone(evt) {
+        this.refs.cloneForm.open();
+    },
+    _onCloneSubmit(streamId, stream) {
+        this.props.onClone(this.props.stream.id, stream);
+    },
+    _onQuickAdd() {
+        this.props.onQuickAdd(this.props.stream.id);
+    },
+    componentDidMount() {
+        UsersStore.load(this.props.username).done((user) => {
```

Comment:
I would actually even do this globally, like we do with the user preferences, as we will probably need the user all over the place and we are already doing a rest call for the preferences each time. Instead of fetching the preferences only, we could just fetch the whole user. This would also help us to have the permissions of the user available and check against them, so we do not have to pass them in in the scala template. What do you think?